tests/util/test_sendPREMISEvent.py

```diff
@@ -1,29 +1,61 @@
+from datetime import datetime
 from urllib.error import URLError
 
-from mock import Mock
+from mock import Mock, patch
 import pytest
 
 from codalib import util
 
 
-def test_is_successful(monkeypatch):
+EVENT = b"""<?xml version="1.0"?>
+<entry xmlns="http://www.w3.org/2005/Atom">
+  <title>20b28e4ee56811e9905154bf64888bf3</title>
+  <id>20b28e4ee56811e9905154bf64888bf3</id>
+  <updated>2019-10-02T17:58:19.982448-05:00</updated>
+  <content type="application/xml">
+    <premis:event xmlns:premis="info:lc/xmlns/premis-v2">
+      <premis:eventIdentifier>
+        <premis:eventIdentifierType>http://purl.org/net/untl/vocabularies/identifier-qualifiers/#UUID</premis:eventIdentifierType>
+        <premis:eventIdentifierValue>20b28e4ee56811e9905154bf64888bf3</premis:eventIdentifierValue>
+      </premis:eventIdentifier>
+      <premis:eventType/>
+      <premis:eventDateTime>2019-10-02T22:58:19</premis:eventDateTime>
+      <premis:eventDetail/>
+      <premis:eventOutcomeInformation>
+        <premis:eventOutcome/>
+      </premis:eventOutcomeInformation>
+      <premis:linkingAgentIdentifier>
+        <premis:linkingAgentIdentifierType>http://purl.org/net/untl/vocabularies/identifier-qualifiers/#URL</premis:linkingAgentIdentifierType>
+        <premis:linkingAgentIdentifierValue/>
+        <premis:linkingAgentRole>http://purl.org/net/untl/vocabularies/linkingAgentRoles/#executingProgram</premis:linkingAgentRole>
+      </premis:linkingAgentIdentifier>
+    </premis:event>
+  </content>
+</entry>
+"""
+
+
+@patch('codalib.util.doWebRequest')
+@patch('codalib.util.uuid.uuid4')
+@patch('codalib.util.uuid.uuid1')
+@patch('codalib.bagatom.xsDateTime_format', return_value='2019-10-02T17:58:19.982448-05:00')
+def test_is_successful(mock_xsdt, mock_uuid1, mock_uuid4, mock_doWebRequest):
     """
     Check that sendPremisEvent returns the response and content.
 
     Verify the conditions will cause no exceptions to be
     raised or caught.
     """
     response = Mock(code=201)
-
     expected = (response, 'Fake content')
-    doWebRequest = Mock(return_value=expected)
-    monkeypatch.setattr('codalib.util.doWebRequest', doWebRequest)
+    mock_doWebRequest.return_value = expected
+    mock_uuid1.return_value.hex = mock_uuid4.return_value.hex = '20b28e4ee56811e9905154bf64888bf3'
 
-    actual = util.sendPREMISEvent(
-        'http://example.com', None, None, None, None)
+    actual = util.sendPREMISEvent('http://example.com', None, None, None, None,
+                                  eventDate=datetime(2019, 10, 2, 22, 58, 19))
```

Comment:
Can you tell me why you have added `eventDate` here, since you are mocking it anyway....so I didn't get your intention of adding it.

Reply:
I only patched` xsDateTime_format` in`bagatom` where it gets used in `wrapAtom` and not in `util` where you can pass in a date for `createPREMISEventXML` as I did. I could change it to patch there too.